src/Shell/Task/ModelTask.php

```diff
@@ -395,17 +441,74 @@ public function findBelongsToMany($model, array $associations)
                 $assocTable = substr($otherTableName, 0, $otherOffset);
             }
             if ($assocTable && in_array($assocTable, $tables)) {
-                $habtmName = $this->_camelize($assocTable);
-                $assoc = [
-                    'alias' => $habtmName,
-                    'foreignKey' => $foreignKey,
-                    'targetForeignKey' => $this->_modelKey($habtmName),
-                    'joinTable' => $otherTableName
-                ];
-                if ($assoc && $this->plugin) {
-                    $assoc['className'] = $this->plugin . '.' . $assoc['alias'];
+                if (!in_array($otherTableName, $tables)) {
+                    continue;
+                }
+                $assocTableObject = TableRegistry::get($otherTableName);
+
+                $thisForeignKey = $primaryKey;
+                foreach ($thisForeignKey as $i => $fieldName) {
+                    if ($fieldName === 'id') {
+                        $thisForeignKey[$i] = Inflector::singularize($model->table()) . '_id';
+                    }
+                }
+                $otherForeignKeys = $assocTableObject->schema()->columns();
+                foreach ($otherForeignKeys as $i => $fieldName) {
+                    if ($fieldName !== 'parent_id' && (
+                        strlen($fieldName) <= 3 || !preg_match('/^.*_id$/', $fieldName))
+                    ) {
+                        unset($otherForeignKeys[$i]);
+                    }
+                }
+                $thisForeignKey = array_values($thisForeignKey);
+                $otherForeignKeys = array_values($otherForeignKeys);
+                $thisForeignKeyIntersection = array_intersect($thisForeignKey, $otherForeignKeys);
+
+                $reverseForeignKey = false;
+                $reverseForeignKeyIntersection = false;
+                $otherForeignKeysDiff = array_diff($otherForeignKeys, $thisForeignKey);
+                foreach ($otherForeignKeysDiff as $reverseForeignKey) {
```

Comment:
Hello, yes it is supported now, from my baked testing code:

### GroupsTable
```php
    public function initialize(array $config)
    {
        parent::initialize($config);

        $this->table('groups');
        $this->displayField('name');
        $this->primaryKey(['id', 'project_id']);

        $this->belongsTo('Projects', [
            'foreignKey' => ['project_id'],
            'joinType' => 'INNER'
        ]);
        $this->belongsToMany('Users', [
            'foreignKey' => ['group_id', 'project_id'],
            'targetForeignKey' => ['project_id', 'user_id'],
            'joinTable' => 'groups_users'
        ]);
    }
```
### GroupsUsersTable
```php
    public function initialize(array $config)
    {
        parent::initialize($config);

        $this->table('groups_users');
        $this->displayField('project_id');
        $this->primaryKey(['project_id', 'group_id', 'user_id']);

        $this->belongsTo('Projects', [
            'foreignKey' => ['project_id'],
            'joinType' => 'INNER'
        ]);
        $this->belongsTo('Groups', [
            'foreignKey' => ['group_id', 'project_id'],
            'joinType' => 'INNER'
        ]);
        $this->belongsTo('Users', [
            'foreignKey' => ['project_id', 'user_id'],
            'joinType' => 'INNER'
        ]);
    }
```
### UsersTable
```php
    public function initialize(array $config)
    {
        parent::initialize($config);

        $this->table('users');
        $this->displayField('name');
        $this->primaryKey(['project_id', 'id']);

        $this->belongsTo('Projects', [
            'foreignKey' => ['project_id'],
            'joinType' => 'INNER'
        ]);
        $this->belongsToMany('Groups', [
            'foreignKey' => ['project_id', 'user_id'],
            'targetForeignKey' => ['group_id', 'project_id'],
            'joinTable' => 'groups_users'
        ]);
        $this->hasOne('Profiles', [
            'foreignKey' => ['project_id', 'user_id']
        ]);
        $this->hasMany('Comments', [
            'foreignKey' => ['project_id', 'user_id']
        ]);
        $this->hasMany('ConventionalComments', [
            'foreignKey' => ['project_id', 'user_id']
        ]);
        $this->hasMany('Tweets', [
            'foreignKey' => ['project_id', 'user_id']
        ]);
    }
```